src/Classes/PassiveTreeView.lua

```diff
@@ -932,28 +932,58 @@ function PassiveTreeViewClass:AddNodeTooltip(tooltip, node, build)
 		tooltip:AddLine(16, ((node.mods[i].extra or not node.mods[i].list) and colorCodes.UNSUPPORTED or colorCodes.MAGIC)..line)
 	end
 
-	if node.sd[1] and node.allMasteryOptions then
+	-- If node is a Mastery node, check if compare tree is on
+	-- If so, check if the left hand tree is unallocated, but the right hand tree is allocated.
+	-- If so, set the node variable to be the node element from the right hand tree and change the Mastery color
+	-- Then continue processing as normal
+	local masteryColor = ""
+	local mNode = node
+	local rightNodeAlloc = self.compareSpec and self.compareSpec.nodes[node.id].alloc or false
+	if node.type == "Mastery" then
+		if not node.alloc and rightNodeAlloc then
+			mNode = self.compareSpec.nodes[node.id]
+			masteryColor = colorCodes.DEXTERITY
+		end
+		-- If allocated on Left, but not Right, match the colour of the other nodes.
+		if node.alloc and not rightNodeAlloc then
+			masteryColor = colorCodes.STRENGTH
```

Comment:
This sets all masteries in a build without more than 1 tree to have all contained text as red

Reply:
Not sure what you trying to show me here ?

Reply:
solved